File: mvvm-test/src/main/java/com/martin/mvvm/ui/adpter/UserAdapter.java
package com.martin.mvvm.ui.adpter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.martin.mvvm.callback.OnUserActionListener;
import com.martin.mvvm.R;
import com.martin.mvvm.db.bean.User;

import java.util.ArrayList;
import java.util.List;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.UserViewHolder> {
    private List<User> users = new ArrayList<>();
    private OnUserActionListener deleteListener;
    private OnUserActionListener modifyListener;

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_user, parent, false);
        return new UserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        User currentUser = users.get(position);
        holder.nameTv.setText(currentUser.getName());
        holder.passwordTv.setText(currentUser.getPassword());
        holder.deleteBtn.setOnClickListener(v -> {
            deleteListener.onAction(currentUser);
        });
        holder.modifyBtn.setOnClickListener(v -> {
            modifyListener.onAction(currentUser);
        });
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    public void setUsers(List<User> users) {
        this.users = users;
        notifyDataSetChanged();
    }

    public void setItemDeleteListener(OnUserActionListener listener) {
        this.deleteListener = listener;
    }

    public void setModifyListener(OnUserActionListener modifyListener) {
        this.modifyListener = modifyListener;
    }

    static class UserViewHolder extends RecyclerView.ViewHolder {
        TextView nameTv;
        TextView passwordTv;
        Button modifyBtn;
        Button deleteBtn;

        public UserViewHolder(@NonNull View itemView) {
            super(itemView);
            nameTv = itemView.findViewById(R.id.tv_name);
            passwordTv = itemView.findViewById(R.id.tv_password);
            modifyBtn = itemView.findViewById(R.id.btn_modify);
            deleteBtn = itemView.findViewById(R.id.btn_delete);
        }
    }
}

File: room-livedata-recyclerview/src/main/java/com/example/rlr/UserAdapter.kt
package com.example.rlr

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class UserAdapter : RecyclerView.Adapter<UserAdapter.UserViewHolder>() {
    private var users: List<User?>? = ArrayList()

    // 👇 这就是 setUsersView() 的实现
    @SuppressLint("NotifyDataSetChanged")
    fun setUsers(users: List<User?>?) {
        this.users = users // 更新内部数据
        notifyDataSetChanged() // 通知 RecyclerView 刷新
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        val itemView: View = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_user, parent, false)
        return UserViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        val currentUser = users?.get(position)
        if (currentUser != null) {
            holder.etName.text = currentUser.name
            holder.etPassword.text = currentUser.password
        }
        if (currentUser != null) {
            holder.etPassword.text = currentUser.password
        }
    }

    override fun getItemCount(): Int {
        return users!!.size
    }

    class UserViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var etName: TextView = itemView.findViewById<EditText>(R.id.tv_name)
        var etPassword: TextView = itemView.findViewById<EditText>(R.id.tv_password)
    }
}
